test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -318,13 +153,12 @@ public void logIfThereAreRunningTasks() throws InterruptedException, IOException
          */
     }
 
-    @AfterClass
-    public static void close() {
-        if (restTestExecutionContext != null) {
-            restTestExecutionContext.close();
-            adminExecutionContext.close();
-            restTestExecutionContext = null;
-            adminExecutionContext = null;
+    private synchronized void closeClients() throws IOException {
+        try {
+            IOUtils.close(client, adminClient);
```

Comment:
It skips nulls. I'm not 100% clear on the history but we use this pattern fairly frequently.